src/Admin/ReaderThemes.php

```diff
@@ -210,17 +227,38 @@ public function get_default_reader_themes() {
 				]
 			);
 
-			if ( ! is_wp_error( $response ) ) {
-				set_transient( $cache_key, $response, DAY_IN_SECONDS );
+			if ( is_array( $response ) ) {
+				$response = (object) $response;
 			}
-		}
 
-		if ( is_wp_error( $response ) ) {
-			return [];
-		}
+			/**
+			 * The response must minimally be an object with a themes array.
+			 *
+			 * @see https://wordpress.org/support/topic/issue-during-activating-the-updated-plugins/#post-13383737
+			 */
+			if (
+				is_wp_error( $response )
+				|| ! is_object( $response )
+				|| ! property_exists( $response, 'themes' )
+				|| ! is_array( $response->themes )
+			) {
+				$this->themes_api_error = new WP_Error(
+					'amp_themes_api_invalid_response',
+					__( 'The request for reader themes from the WordPress.org resulted in an invalid response. Please try again later or contact your host.', 'amp' )
+				);
+				return [];
+			} else {
```

Comment:
The `else` is not needed here, as the `if` already ends in a `return`. This will get rid of one level of indentation.

Basically, both `if`s here are early returns, with the `set_transient()` being the main operation that is left.

Reply:
Couldn't add a suggestion across the entire block, so here's what I mean as a screenshot:

![Image 2020-09-17 at 11 13 23 PM](https://user-images.githubusercontent.com/83631/93528994-8223ea80-f93b-11ea-802e-78a9e7420f2c.png)

Reply:
Good point. Updated in 89a3681ec87083bd0ac28790960a5d31811e33b8